test/Macros/lazy_parsing.swift

```diff
@@ -0,0 +1,39 @@
+// REQUIRES: swift_swift_parser
+
+// This test ensures that we only attempt to parse the syntax tree of a
+// secondary file if the primary file needs it.
+
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/stats-no-lookup)
+// RUN: %empty-directory(%t/stats-lookup)
+// RUN: split-file %s %t
+
+// RUN: %host-build-swift -swift-version 5 -emit-library -o %t/%target-library-name(MacroDefinition) -parse-as-library -module-name=MacroDefinition %S/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
+
+// RUN: %target-swift-frontend -typecheck -swift-version 5 -load-plugin-library %t/%target-library-name(MacroDefinition) -stats-output-dir %t/stats-no-lookup -primary-file %t/b.swift %t/a.swift
+// RUN: %target-swift-frontend -typecheck -swift-version 5 -load-plugin-library %t/%target-library-name(MacroDefinition) -stats-output-dir %t/stats-lookup -primary-file %t/c.swift %t/a.swift
+
+// RUN: %{python} %utils/process-stats-dir.py --evaluate-delta 'ExportedSourceFileRequest > 0' %t/stats-no-lookup %t/stats-lookup
```

Comment:
Am I correct in my understanding that this is checking that there's more `ExportedSourceFileRequest`s in lookup than no lookup? Ie. it's possible we could still have lookups in both? Is it worth doing a check similar to `ensure_no_astgen.swift` for the no lookup case?

Reply:
> Am I correct in my understanding that this is checking that there's more ExportedSourceFileRequests in lookup than no lookup?

Yeah

> it's possible we could still have lookups in both?

I guess it would be possible if we had a rogue SourceFile or something, or the caching of the request was broken, but otherwise the maximum should be 2, right?

I was trying to write this test bearing in mind that the numbers will change based on debug/release due to the round-trip checking. That being said, I realize now it won't work as expected because `process-stats-dir` doesn't handle missing counters well :/ Let me see if I can come up with something better.

Reply:
Figured I should just go ahead and fix the stat printing logic to output 0 counters, so updated this to check the absolute counters for both runs (but we still need to use `<=` to account for difference in debug vs release)